pyserini/search/reranker.py

```diff
@@ -0,0 +1,106 @@
+import enum
+from sklearn.linear_model import LogisticRegression
+from sklearn.svm import SVC
+from typing import List
+from ..vectorizer import TfidfVectorizer
+import uuid
+import os
+
+
+class ClassifierType(enum.Enum):
+    LR = 'lr'
+    SVM = 'svm'
+
+
+class FusionMethod(enum.Enum):
+    AVG = 'avg'
+
+
+class PseudoRelevanceClassifierReranker:
+    def __init__(self, lucene_index: str, clf_type: List[ClassifierType], r=10, n=100, alpha=1):
+        self.r = r
+        self.n = n
+        self.alpha = alpha
+        self.clf_type = clf_type
+
+        if len(clf_type) > 2:
+            raise Exception('Re-ranker takes at most two classifiers')
+
+        self.vectorizer = TfidfVectorizer(lucene_index, min_df=5)
+
+    def set_classifier(self, clf_type: ClassifierType):
+        if clf_type == ClassifierType.LR:
+            self.clf = LogisticRegression()
+        elif clf_type == ClassifierType.SVM:
+            self.clf = SVC(kernel='linear', probability=True)
+        else:
+            raise Exception("Invalid classifier type")
+
+    def get_prf_vectors(self, doc_ids: List[str]):
+        train_docs = doc_ids[:self.r] + doc_ids[-self.n:]
```

Comment:
Add a comment here to explain what you're doing here... notation might not be familiar to all - and line below.
This is actually a cute use of * notation.